api/src/permissions/lib/fetch-policies.ts

```diff
@@ -0,0 +1,39 @@
+import type { Accountability, Filter } from '@directus/types';
+import type { AccessRow } from '../modules/process-ast/types.js';
+import { filterPoliciesByIp } from '../utils/filter-policies-by-ip.js';
+import { withCache } from '../utils/with-cache.js';
+import type { Context } from '../types.js';
+
+export const fetchPolicies = withCache('policies', _fetchPolicies, ({ roles, user, ip }) => ({ roles, user, ip }));
+
+/**
+ * Fetch the policies associated with the current user accountability
+ */
+export async function _fetchPolicies(
+	{ roles, user, ip }: Pick<Accountability, 'user' | 'roles' | 'ip'>,
+	context: Context,
+): Promise<string[]> {
+	const { AccessService } = await import('../../services/access.js');
+	const accessService = new AccessService(context);
+
+	let filter: Filter;
+
+	// No roles and no user means unauthenticated request
+	if (roles.length === 0 && !user) {
```

Comment:
Users without a role no longer assume the Public role. Should this behaviour be changed?

```suggestion
	// Public role is used when the role or user is missing
	if (roles.length === 0 || !user) {
```

Reply:
Updated suggestion to include policies directly attached to users.

```suggestion
	if (roles.length === 0) {
		// Users without role assumes the Public role permissions along with their attached policies
		if (user) {
			filter = { _or: [{ role: { _null: true }, user: { _null: true } }, { user: { _eq: user } }] };
		}
		// No roles and no user means an unauthenticated request
		else {
			filter = { role: { _null: true }, user: { _null: true } };
		}
```